### ISSUE TITLE
Non-zero RELA section contents

### ISSUE THREAD

**Issue by ghost** [nidud/asmc#22]
@Hello71 reported that jwasm-family assemblers (jwasm, asmc, uasm) generate wrong relocation.

See also:
https://sourceforge.net/p/sevenzip/bugs/2338/
https://sourceware.org/pipermail/binutils/2022-April/120428.html

**Comment by nidud** [comment]
Confirmed.

asmc -elf64 test.asm
```
include stdio.inc
.data
a dd 1
b dd 2,3
.code
main proc
    printf( "%d, %d, %d\n", a, b, b[4] )
    ret
main endp
    end
...
test.obj:     file format elf64-x86-64

RELOCATION RECORDS FOR [.text]:
OFFSET           TYPE              VALUE 
0000000000000006 R_X86_64_PC32     b-0x0000000000000004
000000000000000c R_X86_64_PC32     b-0x0000000000000004
0000000000000012 R_X86_64_PC32     a-0x0000000000000004
0000000000000019 R_X86_64_PC32     DS0000-0x0000000000000004
0000000000000020 R_X86_64_PC32     printf-0x0000000000000004

```
gcc -c gtest.c
```
#include <stdio.h>

int a = 1;
int b[2] = { 2, 3 };

int main()
{
	printf("%d, %d, %d\n", a, b[0], b[1]);
	return(0); 
}
...
gtest.o:     file format elf64-x86-64

RELOCATION RECORDS FOR [.text]:
OFFSET           TYPE              VALUE 
0000000000000006 R_X86_64_PC32     b
000000000000000c R_X86_64_PC32     b-0x0000000000000004
0000000000000012 R_X86_64_PC32     a-0x0000000000000004
0000000000000019 R_X86_64_32       .rodata
0000000000000023 R_X86_64_PC32     printf-0x0000000000000004

```

**Comment by nidud** [comment]
Added a fix tor the array issue in v2.33.46
```
Sha256Opt.obj:     file format elf64-x86-64

RELOCATION RECORDS FOR [.text]:
OFFSET           TYPE              VALUE 
000000000000004b R_X86_64_PC32     Reverse_Endian_Mask-0x0000000000000004
0000000000000088 R_X86_64_PC32     K_CONST-0x0000000000000004
00000000000000a1 R_X86_64_PC32     K_CONST+0x000000000000000c
00000000000000be R_X86_64_PC32     K_CONST+0x000000000000001c
00000000000000ed R_X86_64_PC32     K_CONST+0x000000000000002c
000000000000011c R_X86_64_PC32     K_CONST+0x000000000000003c
000000000000014b R_X86_64_PC32     K_CONST+0x000000000000004c
000000000000017a R_X86_64_PC32     K_CONST+0x000000000000005c
00000000000001a9 R_X86_64_PC32     K_CONST+0x000000000000006c
00000000000001d8 R_X86_64_PC32     K_CONST+0x000000000000007c
0000000000000207 R_X86_64_PC32     K_CONST+0x000000000000008c
0000000000000236 R_X86_64_PC32     K_CONST+0x000000000000009c
0000000000000265 R_X86_64_PC32     K_CONST+0x00000000000000ac
0000000000000294 R_X86_64_PC32     K_CONST+0x00000000000000bc
00000000000002c3 R_X86_64_PC32     K_CONST+0x00000000000000cc
00000000000002ee R_X86_64_PC32     K_CONST+0x00000000000000dc
0000000000000307 R_X86_64_PC32     K_CONST+0x00000000000000ec

```

**Comment by Hello71** [comment]
lgtm, thanks!
